src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -1,47 +1,150 @@
 package io.zerobase.smarttracing.resources
 
+import com.fasterxml.jackson.annotation.JsonProperty
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.zerobase.smarttracing.GraphDao
+import io.zerobase.smarttracing.MultiMap
+import io.zerobase.smarttracing.models.OrganizationId
 import io.zerobase.smarttracing.models.IdWrapper
+import io.zerobase.smarttracing.models.SiteResponse
+import io.zerobase.smarttracing.models.ScannableId
 import javax.ws.rs.*
 import javax.ws.rs.core.MediaType
 
-//region Request Models
+/**
+ * Requests from clients.
+ */
+data class Contact(
+    @JsonProperty("phone") val phone: String,
+    @JsonProperty("email") val email: String,
+    @JsonProperty("contactName") val contactName: String
+)
 
-//endregion
+data class Location(
+    @JsonProperty("latitude") val latitude: Float,
+    @JsonProperty("longitude") val longitude: Float
+)
+
+data class CreateOrganizationRequest(
+    @JsonProperty("name") val name: String,
+    @JsonProperty("contact") val contact: Contact,
+    @JsonProperty("address") val address: String,
+    @JsonProperty("hasTestingFacilities") val hasTestingFacilities: Boolean?,
+    @JsonProperty("hasMultipleSites") val hasMultipleSites: Boolean?
+)
+
+data class CreateSiteRequest(
+    @JsonProperty("name") val name: String?,
+    @JsonProperty("category") val category: String,
+    @JsonProperty("subcategory") val subcategory: String,
+    @JsonProperty("location") val location: Location,
+    @JsonProperty("isTesting") val isTesting: Boolean,
+    @JsonProperty("contact") val contact: Contact?
+)
+
+data class CreateScannableRequest(
+    @JsonProperty("type") val type: String,
+    @JsonProperty("singleUse") val singleUse: Boolean
+)
 
 @Path("/organizations")
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
-class OrganizationsResource {
+@SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "false positive")
```

Comment:
what is causing this warning? i've seen it for try-with-resources on java11, but i don't see anything in here that should cause it. also, these annotations should be applied as narrowly as possible. this belongs on the specific function that is causing it, assuming that it can't be avoided

Reply:
createOrganization is coausing it, saying there is a redundant null check of id for some reason